### ISSUE TITLE
Error when using field on a dataclass with __slots__

### ISSUE THREAD

**Issue by Finistere** [microsoft/pyright#3384]
**Describe the bug**
Pyright reports a `reportGeneralTypeIssues` when using `field()` on a `@dataclass` with `__slots__`. `mypy --strict` (version `0.942`) works. It's easy to fix with a custom `__init__()` function though, so not a big deal neither.

**To Reproduce**
Code to typecheck:

```python
from dataclasses import dataclass, field


@dataclass
class Dummy:
    __slots__ = ('values',)
    values: list[int] = field(default_factory=list)

```

**Expected behavior**
No errors

**Screenshots or Code**
```
scratch_1.py
  scratch_1.py:7:5 - error: "values" conflicts with instance variable declared in __slots__ (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
Command line, version `1.1.239` with Python `3.9.11`.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.240, which I just published. It will also be included in the next release of pylance.
